### ISSUE TITLE
OpenID Connect (OIDC) Support

### ISSUE THREAD

**Issue by prvaghasiya** [jeffdgr8/kotbase#27]
Thank you for starting this project. 
Does this support OpenID Connect (OIDC) auth?

**Comment by jeffdgr8** [comment]
Yes, Couchbase Sync Gateway supports OIDC authentication and Kotbase supports the Couchbase Lite replicator API to authenticate with a session ID.

The basic steps are:

1) [Configure your Sync Gateway with your OIDC provider.](https://docs.couchbase.com/sync-gateway/current/authentication-users.html#openid-connect)
2) [Create a session ID with a OIDC token.](https://docs.couchbase.com/sync-gateway/current/rest-api.html#/Session/post_db__session)
3) Get the session ID from the response header or created cookie.
4) Use the session ID to authenticate from the client replicator using [session authentication](https://kotbase.dev/current/remote-sync-gateway/#session-authentication).
5) Refresh the session ID when it expires.

Also see [this tutorial](https://developer.couchbase.com/tutorial-syncgateway-openid-auth) and these blog posts:
- [OAuth 2.0 & OIDC Fundamentals for Authentication & Authorization](https://www.couchbase.com/blog/oauth-2-oidc-fundamentals-authentication-authorization/)
- [OIDC Implicit Flow for Client Authentication in Couchbase Sync Gateway](https://www.couchbase.com/blog/oidc-implicit-flow-client-authentication-couchbase-sync-gateway/)
- [OIDC Authorization Code Flow for Client Authentication in Couchbase Sync Gateway](https://www.couchbase.com/blog/oidc-authorization-code-flow-client-authentication-couchbase-sync-gateway/)
